### PR TITLE
HAAR-868: Get list of searchable roles

### PR DESCRIPTION
**Pull request by ravmoj**
Get list of searchable roles

### CODE REVIEW

File: integration-tests/mockApis/manageusers.js [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
+    { roleCode: 'GLOBAL_SEARCH', roleName: 'Global Search' },
+    { roleCode: 'LICENCE_RO', roleName: 'Licence Responsible Officer' },
+    { roleCode: 'LICENCE_VARY', roleName: 'Licence Vary' },
+  ],
+}) =>
+  getFor({
+    urlPattern: '/externalusers/me/searchable-roles',
+    body: content,
+  })
 module.exports = {
```
Comment (karenmillermoj) [inline integration-tests/mockApis/manageusers.js:16]: If you move this section directly into the module.exports (with the others) you don't need to export it.

File: backend/api/oauthApi.js [hunk 2]
```diff
@@ -130,7 +130,7 @@ const oauthApiFactory = (client, { clientId, clientSecret, url }) => {
     oauthAxios,
     deactivateUser,
     amendUserEmail,
-    searchableRoles,
+    //
```
Comment (karenmillermoj) [inline backend/api/oauthApi.js:133]: Can remove comments

File: integration-tests/mockApis/manageusers.js [hunk 3]
```diff
@@ -1,7 +1,20 @@
 const { getFor, stubJson, getMatchingRequests, stubFor } = require('./wiremock')
 
 const stubUserMeRoles = (roles) => getFor({ urlPattern: '/users/me/roles', body: roles })
+
+const stubAExtSearchableRoles = ({
```
Comment (karenmillermoj) [inline integration-tests/mockApis/manageusers.js:5]: Should stubAExtSearchableRoles be stubExtSearchableRoles?

File: integration-tests/mockApis/manageusers.js (hunk 1 above)
Comment (ravmoj) [reply integration-tests/mockApis/manageusers.js:16]: Moved

File: backend/api/oauthApi.js (hunk 2 above)
Comment (ravmoj) [reply backend/api/oauthApi.js:133]: removed

File: backend/api/oauthApi.js [hunk 4]
```diff
@@ -58,7 +58,7 @@ const oauthApiFactory = (client, { clientId, clientSecret, url }) => {
     })
     return client.get(context, `/api/authuser/search?${query}`).then(processPageResponse(context))
   }
-  const searchableRoles = (context) => get(context, '/api/authuser/me/searchable-roles')
+  // const searchableRoles = (context) => get(context, '/api/authuser/me/searchable-roles')
```
Comment (karenmillermoj) [inline backend/api/oauthApi.js:61]: Does this comment still need removing?

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**